streaming/src/main/scala/org/apache/gearpump/streaming/appmaster/TaskSchedulerImpl.scala

```diff
@@ -17,26 +17,51 @@
  */
 package org.apache.gearpump.streaming
 
+import com.typesafe.config.Config
 import org.apache.gearpump.cluster.scheduler.{Relaxation, Resource, ResourceRequest}
 import org.apache.gearpump.streaming.TaskLocator.{Locality, NonLocality, WorkerLocality}
-import org.apache.gearpump.streaming.TaskSet.TaskLaunchData
+import org.apache.gearpump.streaming.TaskSchedulerImpl.TaskLaunchData
 import org.apache.gearpump.streaming.task.TaskId
 import org.apache.gearpump.util.LogUtil
 import org.slf4j.Logger
 
 import scala.collection.mutable
 
-class TaskSet(appId : Int,  dag : DAG) {
+trait TaskScheduler {
```

Comment:
the methods of this trait looks a bit weird to me. I would imagine a `TaskScheduler` is like 

```scala
trait TaskScheduler {
  def schedule(dag: DAG): Option[TaskLaunchData]
}
```

Also, please add some docs if this is meant to be public interface.